### PR TITLE
Improved markdown formatting

### PR DESCRIPTION
**Pull request by Avasam**
I was modifying a markdown file and noticed that their formatting was all over the place, and that block codes could be made much more readable, so I:
- Used code blocks with language marker (which should add syntax highlighting depending on the viewer, and a copy button on GitHub)
- Ran the files through Markdownlint
  - Which standardized spacing, bare links and bullets within a file.

### CODE REVIEW

File: com/win32comext/mapi/src/mapi_headers/README.md [hunk 1]
```diff
@@ -1,61 +1,67 @@
-These MAPI headers were downloaded from https://www.microsoft.com/en-us/download/details.aspx?id=12905
+These MAPI headers were downloaded from <https://www.microsoft.com/en-us/download/details.aspx?id=12905>
 and enable us to build the MAPI extension in Python 2.7 and up.
 
 The terms for these headers are below:
 
-MICROSOFT SOFTWARE LICENSE TERMS
-MICROSOFT OUTLOOK 2010 MAPI HEADER FILES
+---
```
Comment (mhammond) [inline com/win32comext/mapi/src/mapi_headers/README.md:6]: I like the rest of the PR, but I think copied license text really should remain true to the source.

**Review by mhammond — CHANGES_REQUESTED** [review]

File: com/win32comext/mapi/src/mapi_headers/README.md [hunk 2]
```diff
@@ -7,10 +7,10 @@ MICROSOFT SOFTWARE LICENSE TERMS
 MICROSOFT OUTLOOK 2010 MAPI HEADER FILES
 
 These license terms are an agreement between Microsoft Corporation (or based on where you live, one of its affiliates) and you. Please read them. They apply to the software named above, which includes the media on which you received it, if any. The terms also apply to any Microsoft
-    � updates,
-    � supplements,
-    � Internet-based services, and 
-    � support services
+    • updates,
```
Comment (mhammond) [inline com/win32comext/mapi/src/mapi_headers/README.md:10]: these changes still have snuck in but now seem to be using an odd encoding?

Comment (Avasam) [reply com/win32comext/mapi/src/mapi_headers/README.md:10]: Encoding is likely utf8 given I copy-pasted back. But I can go for a `git checkout upstream/main <file>` to fully restore the file, encoding included.

Comment (mhammond) [reply com/win32comext/mapi/src/mapi_headers/README.md:10]: yeah, maybe the old encoding wasn't utf-8 or something, but we don't expect a diff in that section, right?

**Pull request merged**